### PR TITLE
[MOBILE APP] refactoring profile management

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- Fixes login and sign up issues

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/constants/api.dart [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import 'config.dart';
 
 class AirQoUrls {
-  static String get checkUserExists => Config.airqoApiUserExistsUrl;
-
-  static String get insights =>
-      '${Config.airqoApiUrl}view/measurements/mobile-app/insights';
+  static String get firebaseLookup =>
+      '${Config.airqoApiUrl}users/firebase/lookup';
```
Comment (NoahNsimbe) [inline mobile/lib/constants/api.dart:5]: Endpoint to check if user exists

File: mobile/lib/blocs/profile/profile_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import 'package:app/models/models.dart';
+import 'package:app/services/services.dart';
+import 'package:app/utils/utils.dart';
+import 'package:equatable/equatable.dart';
+import 'package:hydrated_bloc/hydrated_bloc.dart';
+
+part 'profile_event.dart';
+
+class ProfileBloc extends HydratedBloc<ProfileEvent, Profile> {
```
Comment (NoahNsimbe) [inline mobile/lib/blocs/profile/profile_bloc.dart:9]: Hydrated bloc to store user data

File: mobile/lib/screens/auth/phone_auth_widget.dart [hunk 3]
```diff
@@ -283,6 +255,206 @@ class PhoneAuthWidgetState<T extends PhoneAuthWidget> extends State<T> {
...
+              verificationId: state.phoneAuthModel?.verificationId,
+              resendToken: state.phoneAuthModel?.resendToken,
+            );
+            context.read<PhoneAuthBloc>().add(UpdatePhoneAuthModel(
+                  phoneAuthModel,
+                ));
+            context.read<AuthCodeBloc>().add(const UpdateAuthCodeStatus(
+                  status: AuthCodeStatus.success,
+                ));
+          },
```
Comment (NoahNsimbe) [inline mobile/lib/screens/auth/phone_auth_widget.dart:393]: Auto verification @BenjaminSsempala

File: mobile/lib/services/firebase_service.dart [hunk 4]
```diff
@@ -395,54 +385,45 @@ class CloudStore {
...
+      );
 
-      final task = await firebase_storage.FirebaseStorage.instance
-          .ref(docRef)
-          .putFile(file);
+      TaskSnapshot taskSnapshot = await avatarRef.putFile(file);
 
-      return await task.storage.ref(docRef).getDownloadURL();
+      return await taskSnapshot.ref.getDownloadURL();
     } on Exception catch (exception, stackTrace) {
```
Comment (NoahNsimbe) [inline mobile/lib/services/firebase_service.dart:427]: profile picture uploading

File: mobile/lib/services/firebase_service.dart [hunk 5]
```diff
@@ -461,26 +442,46 @@ class CloudMessaging {
...
+    User? user = getUser();
+    if (user != null && user.isAnonymous) {
+      return true;
+    }
+
+    userCredential = await FirebaseAuth.instance.signInAnonymously();
+
+    return userCredential.user != null;
+  }
+
```
Comment (NoahNsimbe) [inline mobile/lib/services/firebase_service.dart:456]: signing in anonymously.

**Review by BenjaminSsempala — APPROVED** [review]
LGTM

File: mobile/lib/screens/auth/phone_auth_widget.dart (hunk 3 above)
Comment (BenjaminSsempala) [reply mobile/lib/screens/auth/phone_auth_widget.dart:393]: Thanks @NoahNsimbe

**Review by Baalmart — APPROVED** [review]
thanks @NoahNsimbe

**Pull request merged**